app/views/directives/osc-secrets.html

```diff
@@ -0,0 +1,62 @@
+<ng-form name="secretsForm" class="osc-secrets-form">
+  <div ng-repeat="pickedSecret in pickedSecrets">
+    <div class="form-group">
+      <div class="row picked-secret">
+        <div ng-class="type === 'Source Secret' ? 'col-sm-6' : 'col-sm-12'">
+          <label class="picker-label" ng-show="$index === 0">{{type | startCase}}</label>
+          <ui-select ng-model="pickedSecret.name">
+            <ui-select-match allow-clear="{{type === 'Source Secret' ? false : true}}" placeholder="Secret name">{{$select.selected}}</ui-select-match>
```

Comment:
You should be able to make the empty option the same height since you control the HTML markup inside the ui-select. I'd also put it at the top.

The `X` button feels really strange for all of the selects.